### ISSUE TITLE
Allow configuring frontend for backend from another provider

### ISSUE THREAD

**Issue by bfncs** [containous/traefik#3695]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I'd like to add a frontend for a backend from another provider. In my particular case I want to implement a blue-green deployment setting with multiple backends from the consul provider and configuring a frontend for one of these backends from the file provider. I can't announce the required frontend through consul itself because I need an instantaneous switch from one backend to another on config reload.

This is somewhat a simpler version of #1122, because there is no need to balance load between backends from different providers but just a direct relation from one frontend to one backend config.

Implementing this would be backwards incompatible if you have unintentional name collisions between backends of different providers (which is improbable). I would expect adding the provider name to the frontend's backend parameter explicitly (or some kind of more global config) would be needed to get around this.

Please let me know what you think about this proposal and how probable it is to get a PR implementing it added.

**Comment by dansteen** [comment]
I would like to see this as well.   In my case we are providing servers via the `consul_catalog` provider, but would like to set dynamic `Host` rules via the `consul` provider.   This would allow us to dynamically change the rules at will via a simple update of consul.   

Thanks!

**Comment by rdgacarvalho** [comment]
I am working hard to try find way to do that as well!
In my case i am using docker swarm and I stopped use file and put all rules for frontend, backend, circuitbreaker and son inside my docker-compose file (crazy and longer) but now I can update the weights, tags, frontEndRules and etc.

I guess the way is use tags.

**Comment by dansteen** [comment]
Yeah, tags definitely make it easier.   In our case we have an application that is started by [nomad](http://nomadproject.io).  We usually set tags that include the routing rules so that our applications respond to the correct urls, however, in this case the application allows the creation of subdomains dynamically inside the app.   So we need to dynamically add in routing rules into the frontend to allow those subdomains to work.  

For now, the subdomains are not added all that often, so we are just going to trigger a redeploy of the nomad app when one is added (which will semi-dynamically add in new routing rules), however, that's really all sorts of awful.

**Comment by marranz** [comment]
+1 In my case i'm trying to create the frontends using dynamodb pointing to a backend which is created through a kubernetes ingress rule.

**Comment by ambernardino** [comment]
Same problem on my side, trying to integrate frontends create via REST with backend created through Docker

**Comment by oastashev** [comment]
+1

**Comment by tanmng** [comment]
Any news on this feature requests gentlemen?

**Comment by kizzie** [comment]
Just throwing in another comment to bump this into people's memories again - I'm trying to do the exact same thing @dansteen is above (swarm containers for the back end but overwriting the frontend with something for only some of the services). Going to use consul tags I guess in the mean time but it makes it all very ugly! Would be nice to have it able to use things cross provider

**Comment by ivanych** [comment]
I also need this feature!

**Comment by mikesir87** [comment]
I, too, have a use case for this feature. We'd like to use a Docker provider for when containers come and go for the backend endpoints. But, would like to use another provider (like a file provider) that defines the frontend rules.

Use case example - using Docker EE provides RBAC capabilities to prevent Team A from updating services that belong to Team B. However, if we rely _only_ on labels for Traefik config, Team A can update its labels to receive traffic intended for Team B. By extracting the frontend rules to another provider (like a file maintained by the container platform team), the teams could no longer modify rules to step on another team's toes.

**Comment by VantuzEE** [comment]
Similar use case on our side, we’d like to create backends through a service fabric provider using file provider to define frontend rules.
